### PR TITLE
Case studies details page 04.04.2023

### LINKED ISSUE #57: Case Study Item
**As a user,**
I want to see a Case Study page, a separate page for case study content. 

### Links to design:
- [Figma web](https://www.figma.com/file/Pp48CPTdPVHY3Fkrxqn45b/Varbin-Tech_?node-id=581-592&t=QoCVi8cvonDsCbqu-4)
- [Figma mob](https://www.figma.com/file/Pp48CPTdPVHY3Fkrxqn45b/Varbin-Tech_?node-id=613-23127&t=QoCVi8cvonDsCbqu-4)

### Requirements:
- The page should use the same layout as in the case of the Index page
- We should use [Dynamic Routes](https://nextjs.org/docs/routing/dynamic-routes)
- Data structure should be the same as in the case of the Index page. We need to extend it and add the necessary fields
- Background colour for the Hero section should be defined via a special prop. It should be optional
- Share buttons shouldn't use any JavaScript. All of them should be plain HTML + CSS
- Instead of a GitHub button, add a Facebook button
- Image wrapper should behave in the same way as in the case of the Index page
- [Technologies should be two columns](https://user-images.githubusercontent.com/3194163/229846358-a49da4ba-9990-402b-8311-87406e227195.jpg)
- [Each section should be a separate component](https://user-images.githubusercontent.com/3194163/229847884-7be3ef76-1c19-4777-97b1-abeba4b7f262.jpg)
- "Book a call" button is a link button which contains a link to Calendly (An example can be found on the Index page)
- "View full case study" button is a link button which contains an MUI link with a Next.js routing component to the next case study

### Attachments:
Technologies should be two columns
Each section should be a separate component

### PR DESCRIPTION
**Pull request by InnaLazarenko**
### List of tasks:
- resolved #57

### CODE REVIEW

File: pages/case-studies/[caseStudyId].tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      socialLink: `https://twitter.com/intent/tweet?text=${pageLink}`,
+      socialIcon: 'twitterIcon',
+      socialBorderRadius: '',
+      socialAriaLabel: 'Share on Twitter',
+    },
+  ];
+
+  return (
+    <>
+      {data ? (
```
Comment (var-bin) [inline pages/case-studies/[caseStudyId].tsx:49]: As far as I see, we can replace all `data ?` with one line check and return `null` in the other case:
```ts
if (data) {
  return (
    //...
  );
}

return null;
```

File: pages/case-studies/index.tsx
Comment (var-bin) [inline pages/case-studies/index.tsx:1]: Please, add a Storybook example

File: components/case-studies/CaseStudyInner.tsx [hunk 2]
```diff
@@ -106,8 +104,7 @@ const CaseStudyInner: FC<CaseStudyInnerProps> = ({ direction, data }) => {
               </>
             ) : null}
 
-            {/* @TODO Uncomment when case study is ready */}
-            {/* <Button href="#">View full case study</Button> */}
+            <ButtonLink href={'/case-studies/' + id}>View full case study</ButtonLink>
```
[the remark this answers is not in the document]
Comment (InnaLazarenko) [reply components/case-studies/CaseStudyInner.tsx:107]: '+

File: components/common/next-page/NextPage.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import Container from '@mui/material/Container';
+import Typography from '@mui/material/Typography';
+import Stack from '@mui/material/Stack';
+
+import { Settings } from '../../../constants/settings';
+
+import Button from '../buttons/Button';
+import ButtonLink from '../buttons/ButtonLink';
+
+const NextPage = () => {
```
[the remark this answers is not in the document]
Comment (InnaLazarenko) [reply components/common/next-page/NextPage.tsx:10]: '+

File: components/common/next-page/NextPage.tsx [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+          padding: '0 20px',
+          maxWidth: '580px',
+          fontSize: { xs: '32px', md: '40px' },
+          lineHeight: '1.2em',
+        }}
+      >
+        Digital Toucan - JQL Search Extensions for Jira
+      </Typography>
+      <Stack direction="row" spacing={3} justifyContent="center">
+        <ButtonLink href="/case-studies/1">View full case study</ButtonLink>
```
[the remark this answers is not in the document]
Comment (InnaLazarenko) [reply components/common/next-page/NextPage.tsx:33]: '+

File: components/common/social-icons/SocialIcons.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,66 @@
+import { storiesOf } from '@storybook/react';
+
+import SocialIcons from './SocialIcons';
+
+const mockSocialIcons = [
```
[the remark this answers is not in the document]
Comment (InnaLazarenko) [reply components/common/social-icons/SocialIcons.stories.tsx:5]: '+

File: pages/case-studies/index.tsx
Comment (InnaLazarenko) [reply pages/case-studies/index.tsx:1]: '+

File: pages/case-studies/[caseStudyId].tsx
[the remark this answers is not in the document]
Comment (InnaLazarenko) [reply pages/case-studies/[caseStudyId].tsx:1]: '+

**Pull request merged**